thetis/callback.py

```diff
@@ -44,6 +44,7 @@ def add(self, callback, mode):
         :arg mode: register callback under this mode
         :type mode: string
         """
+        assert isinstance(mode, (str, unicode))
```

Comment:
In python 2 AKA: `isinstance(mode, basestring)`. I suppose we need to revisit this when we're switching to python 3.